src/source/vector_tile_source.js

```diff
@@ -73,8 +74,8 @@ class VectorTileSource extends Evented implements Source {
     }
 
     setBounds(bounds?: [number, number, number, number]) {
-        this.bounds = bounds;
         if (bounds) {
```

Comment:
I think the intent might have been to allow `setBounds(null)` or `setBounds(undefined)` to remove an existing bounds. It doesn't look like that's actually used anywhere, however, and `setBounds` isn't a public API. I'd say we should just inline `setBounds` where it's used in the `loadTileJSON` in `load`, and set the bounds in the initial source properties where they're needed in `{raster,vector}_tile_source.test.js`.

Relatedly, I don't think it should be necessary to add any calls to `setBounds`, as you've done above. `loadTileJSON` calls the callback in both cases: when the initial source properties contain inline TileJSON properties, and when they contain a `url` to TileJSON.